### ISSUE TITLE
Webpack executor minimizes classnames when optimization: true

### ISSUE THREAD

**Issue by Sylphide** [nrwl/nx#12706]
## Current Behavior
Webpack executor minimizes classnames when optimization is set to true.
It was fixed in this #8537
This happens since [normalize-options.ts](https://github.com/nrwl/nx/blob/master/packages/webpack/src/executors/webpack/lib/normalize-options.ts) transforms optimization option from `true` to `{scripts: true, styles: true}`.

## Expected Behavior
Optimization options should work as intended in #8537

### Environment
   Node : 16.13.0
   OS   : linux x64
   yarn : 1.22.17
   
   nx : 15.0.0
   @nrwl/angular : Not Found
   @nrwl/cypress : 15.0.0
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.0.0
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.0.0
   @nrwl/expo : Not Found
   @nrwl/express : 15.0.0
   @nrwl/jest : 15.0.0
   @nrwl/js : 15.0.0
   @nrwl/linter : 15.0.0
   @nrwl/nest : 15.0.0
   @nrwl/next : 15.0.0
   @nrwl/node : 15.0.0
   @nrwl/nx-cloud : 14.7.0
   @nrwl/nx-plugin : 15.0.0
   @nrwl/react : 15.0.0
   @nrwl/react-native : Not Found
   @nrwl/rollup : 15.0.0
   @nrwl/schematics : Not Found
   @nrwl/storybook : 15.0.0
   @nrwl/web : 15.0.0
   @nrwl/webpack : 15.0.0
   @nrwl/workspace : 15.0.0
   typescript : 4.8.4

**Comment by jaysoo** [comment]
Yeah this is intended since optimization for styles means minifying the class names. We don't recommend turning this off for production, but you want to you can pass `{ scripts: true, styles: false }` in your `project.json` file.

**Comment by Sylphide** [comment]
I don't really understand what is the link between styles optimization and
class names. I use the webpack executor in a Nestjs project and need the
classnames to keep their names for the swagger cli plugin to work properly
in production. If I understood correctly, styles optimization is only
relevant if target = 'web'

For target = 'node', since the optimization configuration goes from true to
{scripts: true, styles: true}, I guess those following lines are useless
since optimization will never be equal to true

```
webpack/src/utils/config.ts l.157
...
if (options.optimization === true) {
webpackConfig.optimization = {
minimize: true,
minimizer: [
new TerserPlugin({
terserOptions: {
mangle: false,
keep_classnames: true,
},
}),
],
concatenateModules: true,
};
}
```

**Comment by JonnyBGod** [comment]
This also breaks Nestjs Graphql code first.
The schema generation in production does not work as classes changed names and also use unsupported character for GraphQL.

**Comment by abonifacio** [comment]
Experiencing the same issue. Looks like the regression was introduced in this PR #11966 . @jaysoo can we reopen?

https://github.com/nrwl/nx/pull/11966/files#diff-b4c8c0102bc43532dd1ef9440834c7e2351e5dfed40cfe4a26437ed126c6a49cL27-L34

**Comment by abonifacio** [comment]
Experiencing the same issue. This PR https://github.com/nrwl/nx/pull/12221/files fixes it. @jaysoo can we keep this open?

**Comment by mklinke** [comment]
> Introduced in https://github.com/nrwl/nx/pull/11966/files#diff-891ed1c6cd733eaa28d9f44efe10195ee093b21f3fec16330217ef78d21799cdR28-R34

I found this issue while trying to upgrade from 14.7.5 to the latest 14.x minor version (can't upgrade to 15.x yet due to some Angular dependencies). 

It looks like the referenced commit was pushed briefly after 14.7.5, but hasn't been fixed on the 14.8.x branch. 

Is there a chance the fix will be applied to that branch, too? I've looked into the Git history and it seems like most fixes for 15.x have been almost immediately cherry-picked to 14.8.x. 

Is there any particular reason this hasn't been cherry-picked to 14.8.x?

Just trying to understand. Thanks in advance for any answer.
